ADR/0039-workspace-deprecation.md

```diff
@@ -0,0 +1,98 @@
+cloud native apps ecosystem. We want everything to work on vanilla Kubernetes - no OpenShift required.
+
+* We had our [first upstream Konflux release](https://github.com/konflux-ci/konflux-ci) which doesn't use [kubesaw] or relying
+on the Workspace abstraction, and is deployable on vanilla Kubernetes.
+
+* We made a decision (now long ago) to drop [KCP] from our architecture, which initially introduced the workspace concept. Both transparent multi-cluster as a feature and workspaces as a concept came from KCP.
+
+* We made the decision (at Red Hat) to stop trying to offer a managed service interface for Konflux. The user signup, waitlist, and workspace provisioning interfaces made a lot of sense there.
+
+## Decision
+
+1. We will remove the Workspace abstraction from Konflux and use standard Kubernetes namespaces
+directly for providing a place for users to run their pipelines and store their configurations
+and secrets.
+
+2. We will stop using the term Workspace, and start to use the term Namespace.
+
+3. The Konflux UI will expose a wizard for creating a new namespace. This
+wizard will be available to any user that has permissions to create namespaces.
+
+4. Konflux will provide an optional automation for creating a namespace for a user
+when he/she logins for the first time.
+
+5. Konflux won't be opinionated about the mechanism for initializing/maintaining namespaces with
+supporting resources such as ResourceQuota and LimitRange. Konflux will differ
+this responsibility to other tools which are specialized in this task such as 
+(but not limited to) [Kyverno](https://kyverno.io/policies/best-practices/add-ns-quota/add-ns-quota/) and [ArgoCD](https://github.com/konflux-ci/namespace-generator).
+
+6. Konflux will provide a thin [backend service](https://github.com/konflux-ci/workspace-manager) for listing the namespaces where the user has at least view access
+to the Konflux CRDs. This list will be used by the namespace switcher in the UI.
+This is required since the Kubernetes API doesn't let the user to list a subset
+of namespace. The user gets permissions to list all namespace or none.
+
+7. Konflux will provide ClusterRoles that will grant permissions to the Konflux
+and Tekton CRDs. Those will be (aggregated)[https://github.com/konflux-ci/konflux-ci/issues/440] to the built-in Kubernetes roles (`view`, `edit`, `admin`).
+
+8. Public viewer access will be provided by assigning the `view` role (see above) to the
+`system:authenticated` built-in group that contains all the authenticated users.
+
+9. Same as Kubernetes, Konflux won't have a resource for representing a user. Instead,
+it will use external Identity providers.
+
+9. Konflux won't provide a way for creating a ephemeral namespaces. It will differ this
+task to another tool.
+
+10. The `join the waitlist` button will be removed from the Konflux UI.
+
+11. Konflux won't provide a transparent multi-cluster deployment.
```

Comment:
| If a someone wants to use konflux in a multi cluster setting they should be able to without affecting the end-user experience

what does it mean? nothing stops anyone from deploying multiple Konflux instances.

Reply:
When you required this you are basically saying we are not only saying that we will implement a single cluster Konflux deployment, you are also saying there is no room to manage multiple cluster with single Konflux deployment. When there multi cluster management tools out there that could do it.

Reply:
Sorry, maybe I didn't express my thoughts properly. I'm saying that I prefer that Konflux core will be a single cluster deployment. Nothing stops anyone from developing add-ons  or integration with existing tools that will allow Konflux (or any other applications, since mutli-cluster deployment isn't a Konflux specific problem) to have transparent multi-cluster deployment and tenancy model.

Reply:
This is already possible with KubeSaw - you can use single-cluster only (and extend that to multi-cluster later if needed). 
Single-cluster is also used in dev/e2e environments

Reply:
If the are no objections I'll close this thread.